examples/action_manager.py

```diff
@@ -0,0 +1,75 @@
+from random import shuffle
+
+import numpy as np
+from skimage import data
+
+import napari
+from napari._qt.widgets.qt_viewer_buttons import QtViewerPushButton
+from napari.components import ViewerModel
+from napari.utils.action_manager import action_manager
+
+
+def rotate45(viewer):
+    """
+    Rotate layer 0 of the viewer by 45º
+
+    Parameters
+    ----------
+    viewer
+        active (unique) instance of the naparia viewer
+
+    Notes
+    -----
+    The `viewer` parameter needs to be named `viewer`, the action manager will
```

Comment:
is it also possible to annotate it with `napari.Viewer` so that the action manager can infer this as well? also, if there's any relevant documentation about how inference works, could you link to it?

Reply:
There is no real inference right now, anyone can just do `action_manager.context[name] = object`, and any parameter  that match `name` will be given the action roughtly `callback(**action_manager.context)`.

We can make use of the type annotation to check that the parameter are of the right values – that's a good idea, but let's keep that for a subsequent PR.